---
output:
  md_document:
    variant: markdown_github
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "",
  fig.path = "README/README-",
  fig.align = 'center',
  fig.retina=2, fig.width = 7, fig.height = 5,
  warning = FALSE,
  message = FALSE
)
options("width"=200)
```

iNEXT.link (R package)
=====

<h5 align="right">Latest version: `r Sys.Date()`</h5>

<font color="394CAE"><h3 color= 394CAE style = "font-weight: bold"> Introduction to iNEXT.link (R package): Excerpt from iNEXT.link UserGuide </h3> </font>
<br>
<h5><b>Anne Chao, K.S. Hu, K.W. Chen, C.G. Lo, S.Y. Wang</b>
<br><br>
<i>Institute of Statistics, National Tsing Hua University, Hsin-Chu, Taiwan 30043</i>  
</h5>
<br>
`iNEXT.link` is an R package that extends the concepts of iNEXT.3D, iNEXT.4step and iNEXT.beta3D to ecological networks. In this document, we provide a brief overview of `iNEXT.link` and its functionalities. Detailed information about the `iNEXT.link` functions can be found in the `iNEXT.link` manual, which is also available on [Github](https://github.com/AnneChao). For users without an R background, an online version("https://chao.shinyapps.io/iNEXT_link/") of `iNEXT.link` is also available.

`iNEXT.link` is primarily designed to calculate and analyze various measures of diversity in ecological networks. Specifically, the package calculates three Hill numbers of order q (species richness, Shannon diversity, and Simpson diversity), as well as phylogenetic and functional diversity levels.

For single ecological networks, `iNEXT.link` provides tools for analyzing diversity. The package provides two types of rarefaction and extrapolation (R/E) sampling curves to estimate diversity and confidence intervals for single ecological networks. These include sample-size-based (or size-based) R/E curves and coverage-based R/E curves.

Moreover, `iNEXT.link` offers dissimilarity-turnover curves for the coverage-based R/E curves for gamma, alpha, and beta diversity measures, which can be used to compare diversity patterns across different ecological networks.


### SOFTWARE NEEDED TO RUN INEXT.3D IN R

-   Required: [R](http://cran.rstudio.com/)
-   Suggested: [RStudio IDE](http://www.rstudio.com/ide/download/)

## HOW TO RUN iNEXT.link:

The `iNEXT.link` package can be downloaded from Anne Chao's [iNEXT.link_github](https://github.com/AnneChao/iNEXT.link) using the following commands. For a first-time installation, additional visualization extension packages (`ggplot2`) from CRAN and (`iNEXT.3D`), (`iNEXT.4steps`), and (`iNEXT.beta3D`) from Anne Chao's github must be installed and loaded. 

```{r install, eval=FALSE}
## install the latest version from github
install.packages('devtools')
library(devtools)

# install_github('AnneChao/iNEXT.3D')
# install_github('AnneChao/iNEXT.4steps')
# install_github('AnneChao/iNEXT.beta3D')

install_github('AnneChao/iNEXT.link')

## import packages
library(iNEXT.link)
```


In this document, we provide a quick introduction demonstrating how to run the package `iNEXT.link`(iNterpolation and EXTrapolation in Network diversity). `iNEXT.link` has several main functions: 

## For Single community:

- **iNEXT.link** : Computes taxonomic, phylogenetic, and functional diversity estimates and sample coverage estimates.

- **DataInfo.link** : exhibits basic data information
    
- **estimateD.link** : computes species diversity with a particular user-specified level of sample size or sample coverage.
    
- **AO.link**:compute asymptotic (or observed) diversity of order q.

- **Completeness.link** : Calculates estimated sample completeness with order q. 

- **Spec.link** : Computes specialization estimation (observed) with order q.

## For Multi-community:

- **iNEXTbeta.link** : Computing standardized gamma, alpha, beta diversity, and four dissimilarity-turnover indices for three dimensions: taxonomic, phylogenetic and functional diversity at specified sample coverage or sample size.


## Visualizing Results:

- **ggCompleteness.link** : Visualizing the output from the function `Completeness.link`

- **ggSpec.link** : Visualizing the output from the function `Spec.link`

- **ggAO.link** : Visualizing the output from the function `AO.link`

- **ggiNEXT.link** : Visualizing the output from the function `iNEXT.link`
    
- **ggiNEXTbeta.link** : Visualizing the output from the function `iNEXTbeta.link`



First, we load data from `iNEXT.link`:

```{r include=FALSE}
beetles = iNEXT.link::beetles_plotA
beetles = list(Closed = beetles$Closed, Open = beetles$Open)
beetles = lapply(beetles, as.data.frame)
beetles_col_tree = iNEXT.link::beetles_col_tree
beetles_col_distM = iNEXT.link::beetles_col_distM
```


### SINGLE COMMUNITY FUNCTION: iNEXT.link()

We first describe the main function `iNEXT.link()` with default arguments: 

<br>
iNEXT.link(data,diversity = "TD", q = c(0, 1, 2), size = NULL, nT = NULL,
           endpoint = NULL, knots = 40, conf = 0.95, nboot = 30, 
           row.tree = NULL, col.tree = NULL, PDtype = "meanPD", 
           row.distM = NULL, col.distM = NULL, FDtype = "AUC", FDtau = NULL)
<br>

The arguments of this function are briefly described below, and will be explained in more details by illustrative examples in later text.This main function computes diversity estimates of order q, the sample coverage estimates and related statistics for K (if `knots = K`) evenly-spaced knots (sample sizes) between size 1 and the `endpoint`, where the endpoint is described below. Each knot represents a particular sample size for which diversity estimates will be calculated. By default, endpoint = double the reference sample size (total sample size for abundance data). For example, if `endpoint = 10`, `knot = 4`, diversity estimates will be computed for a sequence of samples with sizes (1, 4, 7, 10). 


This function returns an "iNEXT.link" object which can be further used to make plots using the function ggiNEXT.link() to be described below.

```{r, echo=FALSE,warning=FALSE}
Des <- c("data"," a list of data.frames, each data.frames represents col.species-by-row.species abundance matrix.",
"diversity"," selection of diversity type: 'TD' = Taxonomic diversity, 'PD' = Phylogenetic diversity, and 'FD' = Functional diversity.",
"q","	a numerical vector specifying the diversity orders. Default is c(0, 1, 2).",
"size","an integer vector of sample sizes for which diversity estimates will be computed. If NULL, then diversity estimates will be calculated for those sample sizes determined by the specified/default endpoint and knots.",
"endpoint","an integer specifying the sample size that is the endpoint for R/E calculation; If NULL, then endpoint=double the reference sample size;",
"knots","	an integer specifying the number of equally-spaced knots between size 1 and the endpoint. Default is 40.",
"conf","	a positive number < 1 specifying the level of confidence interval. Default is 0.95.",
"nboot"," a positive integer specifying the number of bootstrap replications when assessing sampling uncertainty and constructing confidence intervals. Enter 0 to skip the bootstrap procedures. Default is 30.",
"row.tree","(required only when diversity = 'PD' a phylogenetic tree of row assemblage in the pooled network row assemblage. ",
"col.tree","	(required only when diversity = 'PD') a phylogenetic tree of column assemblage in the pooled network column assemblage. ",
"PDtype"," (required only when diversity = 'PD'), select PD type: PDtype = 'PD' (effective total branch length) or PDtype = 'meanPD' (effective number of equally divergent lineages). Default is 'meanPD', where meanPD = PD/tree depth.",
"row.distM"," (required only when </code>diversity = 'FD') a species pairwise distance matrix for all species of row assemblage in the pooled network row assemblage. ",
"col.distM"," (required only when diversity = 'FD') a species pairwise distance matrix for all species of column assemblage in the pooled network column assemblage.",
"FDtype"," (required only when diversity = 'FD'), select FD type: FDtype = 'tau_values' for FD under specified threshold values, or FDtype = 'AUC' (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is 'AUC'.",
"FDtau"," (required only when diversity = 'FD' and FDtype = 'tau_values'), a numerical vector between 0 and 1 specifying tau values (threshold levels). If NULL (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy)."
)
output <- 
  matrix(Des, 
         ncol=2, byrow = TRUE)

library(htmlTable)
htmlTable(output,
          header =  c("Argument","Description"),align = "l"
          )
```


## DATA FORMAT/INFORMATION
Supported Data Types:

Individual-based abundance data : Input data for each assemblage/site include samples species abundances in an empirical sample of n individuals ("reference sample"). When dealing with N assemblages, the input data consists of N lists of species abundances  


## RAREFACTION/EXTRAPOLATION VIA EXAMPLES

The data sets (tree-beetles interaction data ) are included in iNEXT.link package. The experiment took place in the Steigerwald forest in Germany, where deadwood objects from six tree species were exposed in open and closed habitats. Saproxilic beetles were sampled using stem emergence traps and classified according to their functional traits. Data from four years were pooled for each plot and habitat, and pairwise distances were computed from the Gower distance. Here, the demonstration only uses data from plot A.  For these data, the following commands display the sample species abundances and run the `iNEXT.link()` function for three types of diversty  (`"TD"`, `"PD"`, `"FD"` with threshold dmean, `"AUC"` which integate FD from threshold 0-1) in `q = 0`. 

If one diversity class required, then `iNEXT.link()` function returns including: `$Info` for summarizing data information; `$iNextEst` for showing diversity estimates along with related statistics for a series of rarefied and extrapolated samples; and `$AsyEst` for showing asymptotic diversity estimates along with related statistics, otherwise, returns lists which length equal to the number of diversity class required and also named by diversity class. Among each list include three data frames:`$Info`, `$iNextEst` and `$AsyEst`. 
  
`$Info`, as shown below, returns basic data information.  It can also be presented using function `DataInfo()` (for "TD"), `PDInfo()` (for "PD") and `AUCInfo()` (for "FD").

Because the three kinds of diversity output are similar therefore the demo shows only TD.



```{r setup, include=FALSE, echo=FALSE}
knitr::opts_chunk$set(comment="", message=FALSE, warning=FALSE)
library(iNEXT.link)
library(ggplot2)
```

```{r echo=TRUE}
linkoutTD = iNEXT.link(data = beetles,diversity = 'TD', q = c(0,1,2),nboot = 30)
linkoutTD$DataInfo
```

Diversity estimates and related statistics are computed for these 40 knots, which locates the reference sample at the midpoint of the selected knots. If the argument se=TRUE, then the bootstrap method is applied to obtain the 95% confidence intervals for each diversity and sample coverage estimates. 

For the sample size corresponding to each knot, the first data frame of list `$iNextEst` (as shown below for "size_based") under each diversity class includes the sample size (`m`, i.e., each of the 40 knots), the method (`interpolated`, `observed`, or `extrapolated`, depending on whether the size `m` is less than, equal to, or greater than the reference sample size), the diversity order, the diversity estimate of order q (`qD`, `qPD`, `qFD` and `qAUC`), the 95% lower and upper confidence limits of diversity (`qD.LCL` with`qD.UCL`, `qPD.LCL` with `qPD.UCL`, `qFD.LCL` with `qFD.UCL` and `qAUC.LCL` with `qAUC.UCL`), and the sample coverage estimate (`SC`) along with the 95% lower and upper confidence limits of sample coverage (`SC.LCL`, `SC.UCL`). These sample coverage estimates with confidence intervals are used for plotting the sample completeness curve.


```{r}
head(linkoutTD$iNextEst$size_based)
```

The second data frame of list `$iNextEst` (as shown below for "coverage_based") under each class includes real sample coverage ("SC"), sample size (`m`, i.e., each of the 40 knots), the method (`Rarefaction`, `Observed`, or `Extrapolation`, depending on whether the size `m` is less than, equal to, or greater than the reference sample size),the diversity estimate of order q (qD, qPD, qFD and qAUC) conditioning on "SC", the 95% lower and upper confidence limits of diversity (`qD.LCL` with `qD.UCL`, `qPD.LCL` with `qPD.UCL`, `qFD.LCL` with `qFD.UCL` and `qAUC.LCL` with `qAUC.UCL`). These sample coverage estimates with confidence intervals are used for plotting the coverage-based R/E curves.


```{r}
head(linkoutTD$iNextEst$coverage_based)
```

`$AsyEst` lists the observed diversity, asymptotic estimates, estimated bootstrap s.e. and 95% confidence intervals for Hill numbers with q = 0(`Species richness`), 1(`Shannon diversity`), and 2(`Simpson diversity`). The estimated asymptotes and the observed diversity are calculated via the functions `AO.link()` . The output for the dunes data is shown below. All row and column variables are self-explanatory. 


```{r}
head(linkoutTD$AsyEst)
```


## GRAPHIC DISPLAYS: FUNCTION ggiNEXT.link()

The function `ggiNEXT.link()`, which extends `ggplot2` with default arguments, is described as follows: 

<br>
ggiNEXT.link(outcome, type = 1:3,facet.var = "Assemblage", color.var = "Order.q")  
<br>

Here `outcome` is the object of `iNEXT.link()`'s output. Three types of curves are allowed for different diversity classes:  

(1) Sample-size-based R/E curve (`type=1`): see Figs. 1a and 2a in the main text. This curve plots diversity estimates with confidence intervals as a function of sample size up to double the reference sample size, by default, or a user-specified `endpoint`.  

(2) Sample completeness curve (`type=2`) with confidence intervals: see Figs. 1b and 2b in the main text. This curve plots the sample coverage with respect to sample size for the same range described in (1).  

(3) Coverage-based R/E curve (`type=3`): see Figs. 1c and 2c in the main text. This curve plots the diversity estimates with confidence intervals as a 
function of sample coverage up to the maximum coverage obtained from the maximum size described in (1). 

The argument `facet.var=("Order.q", "Assemblage")` is used to create a separate plot for each value of the specified variable. For example, the following code displays a separate plot (in Figs 1a and 1c) for each value of the diversity order q. The `ggiNEXT.link()` function is a wrapper around 
`ggplot2` package to create a R/E curve using a single line of code. The resulting object is of class `"ggplot"`, so can be manipulated using the `ggplot2` tools. 


The argument `facet.var="Assemblage"` in `ggiNEXT.link` function creates a separate plot for each assembalge, therefore the different Order.q will seperated by different colours as shown below: 

```{r }
# Sample-size-based R/E curves, separating by "assemblage""
ggiNEXT.link(linkoutTD, type = 1, facet.var = "Assemblage")
```

The argument `facet.var="Order.q"` in `ggiNEXT.link` function creates a separate plot for each order, therefore six assemblages will be seperated by different colours as shown below: 
```{r }
# Sample-size-based R/E curves, separating by "Order.q"
ggiNEXT.link(linkoutTD, type = 1, facet.var = "Order.q")
```


The following commands return the sample completeness curve in which different colors are used for the six assemblages. Since the sample completeness curve are same for differnet class of diversity, `ggiNEXT.link` returns only one plot: 

```{r}
ggiNEXT.link(linkoutTD, type = 2, facet.var = "Order.q", color.var="Assemblage")
```


The following commands return the coverage-based R/E sampling curves in which different colors are used for the six assemblages (`facet.var="Assemblage"`) and for three orders(`facet.var="Order.q"`)

```{r}
ggiNEXT.link(linkoutTD, type = 3, facet.var="Assemblage")
```

```{r}
ggiNEXT.link(linkoutTD, type = 3, facet.var="Order.q")
```


## DATA INFORMATION FUNCTION: DataInfo.link()

We can supply the function

<br>
DataInfo.link(data, diversity = "TD", row.tree = NULL, 
              col.tree = NULL, row.distM = NULL, col.distM = NULL) 
<br>

to compute three type diversity('TD','PD','FD') data information, which including sample size, observed species richness, sample coverage estimate, and the first ten abundance frequency counts, and so on.

```{r}
DataInfo.link(beetles, diversity = 'TD')
```

```{r}
DataInfo.link(beetles, diversity = 'PD', col.tree = beetles_col_tree)
```

```{r}
DataInfo.link(beetles, diversity = 'FD', col.distM = beetles_col_distM)
```


## POINT ESTIMATION FUNCTION: estimateD.link()
We also supply the function 

<br>
estimateD.link(data, diversity = "TD", q = c(0, 1, 2),  
               base = "coverage", level = NULL, nboot = 50, conf = 0.95, 
               PDtype = "meanPD", row.tree = NULL, col.tree = NULL, row.distM = NULL, 
               col.distM = NULL, FDtype = "AUC", FDtau = NULL) 
<br>


to compute three type diversity('TD','PD','FD') estimates with q = 0, 1, 2 for any particular level of sample size (`base="size"`) or any specified level of sample coverage (`base="coverage"`) for abundance data . If `level=NULL`, this function computes the diversity estimates for the minimum sample size/coverage among all assemblages. 

For example, the following command returns the taxonomic diversity ('TD') with a specified level of sample coverage of 70% for the tree-beetles interaction data. For some assemblages, this coverage value corresponds to the rarefaction part whereas the others correspond to extrapolation, as indicated in the method of the output. 

```{r}
estimateD.link(beetles, diversity = 'TD', q = c(0,1,2), base = "coverage",level = 0.7)
```

## ASYMPTOTIC AND OBSERVED DIVERSITY FUNCTION: AO.link

<br>
AO.link(data, diversity = "TD", q = seq(0, 2, 0.2),  nboot = 30, 
        conf = 0.95, method = c("Asymptotic", "Observed"), 
        row.tree = NULL, col.tree = NULL, PDtype = "meanPD", row.distM = NULL, 
        col.distM = NULL, FDtype = "AUC", FDtau = NULL)
<br>

The function `AO.link()` can compute three type diversity('TD','PD','FD'),which including empirical diversity and asymptotic diversity. For any specified level of q can be compute.

For example, the following command returns an empirical taxonomic diversity('TD') and asymptotic taxonomic diversity('TD') for dunes data, along with its confidence interval, for a specified q level from 0 to 2.

```{r}
out1 <- AO.link(beetles, diversity = 'TD', q = seq(0, 2, 0.2), method = c("Asymptotic", "Observed"),
                nboot = 5,conf = 0.95)

out1
```


## GRAPHIC DISPLAYS FUNCTION: ggAO.link()

Plots q-profile, time-profile, and tau-profile based on the outcome of AO.link using the ggplot2 package.

The function `ggAO.link()`, which extends `ggplot2` with default arguments, is described as follows: 

<br>
ggAO.link(outcome)
<br>

Here `outcome` is the object of `AO.link`'s output .

```{r }
# q profile curve""
ggAO.link(out1)
```


## SINGLE COMMUNITY FUNCTION: Completeness.link()

Function `Completeness.link()` provides a easy way to compute estimated sample completeness with order q. It has default arguments: 
<br>
Completeness.link(data, q = seq(0, 2, 0.2), nboot = 30, conf = 0.95) 
<br>


## GRAPHIC DISPLAYS FUNCTION:  ggCompleteness.link()

We also provides a realized function `ggCompleteness.link` to plot the output from `Completeness.link()`:

<br>ggCompleteness.link(output)
<br>


We use data beetles to calculate sample completeness and plot it.

```{r fig.height=8, out.width="70%"}
out1 <- Completeness.link(data = beetles)
ggCompleteness.link(out1)
```



## SINGLE COMMUNITY FUNCTION: Spec.link()

We describe the main function `Spec.link()` with default arguments: 

<br>
Spec.link(data, q = seq(0, 2, 0.2), method = "Estimated",
         nboot = 30, conf = 0.95, E.class = c(1:5), C = NULL) 
<br>


## GRAPHIC DISPLAYS FUNCTION: ggSpec.link()

We provide a function `ggSpec.link()` to plot the output from `Spec.link()`.

<br>
ggSpec.link(output)
<br>


There are an example for funciton `Spec.link` and function `ggSpec.link`. 

```{r}
out1 <- Spec.link(data = beetles)
ggSpec.link(out1)
```



### Multi-community

## MULTI-COMMUNITY FUNCTION: iNEXTbeta.link()

```{r eval=FALSE}
iNEXTbeta.link(data, diversity = "TD", level = seq(0.5, 1, 0.05), 
               q = c(0, 1, 2), nboot = 20, conf = 0.95, 
               PDtype = "meanPD", row.tree = NULL, col.tree = NULL, row.distM = NULL, 
               col.distM = NULL, FDtype = "AUC", FDtau = NULL, FDcut_number = 30) 
```


```{r, echo=FALSE,warning=FALSE}
Des <- c("data"," data can be input as a matrix/data.frame (species by assemblages), or a list of matrices/data.frames, each matrix represents species-by-assemblages abundance matrix",
"diversity"," selection of diversity type: 'TD' = Taxonomic diversity, 'PD' = Phylogenetic diversity, and 'FD' = Functional diversity.",
"level","a sequence specifying the particular sample coverages (between 0 and 1). Default is seq(0.5, 1, 0.05).",
"q","	a numerical vector specifying the diversity orders. Default is c(0, 1, 2).",
"nboot"," a positive integer specifying the number of bootstrap replications when assessing sampling uncertainty and constructing confidence intervals. Bootstrap replications are generally time consuming. Enter 0 to skip the bootstrap procedures. Default is 30.",
"conf","a positive number < 1 specifying the level of confidence interval. Default is 0.95.",
"PDtype"," (required only when diversity = 'PD'), select PD type: PDtype = 'PD' (effective total branch length) or PDtype = 'meanPD' (effective number of equally divergent lineages). Default is 'meanPD', where meanPD = PD/tree depth.",
"row.tree","(required only when diversity = 'PD' a phylogenetic tree of row assemblage in the pooled network row assemblage. ",
"col.tree","	(required only when diversity = 'PD') a phylogenetic tree of column assemblage in the pooled network column assemblage. ",
"row.distM"," (required only when </code>diversity = 'FD') a species pairwise distance matrix for all species of row assemblage in the pooled network row assemblage. ",
"col.distM"," (required only when diversity = 'FD') a species pairwise distance matrix for all species of column assemblage in the pooled network column assemblage.",
"FDtype"," (required only when diversity = 'FD'), select FD type: FDtype = 'tau_values' for FD under specified threshold values, or FDtype = 'AUC' (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is 'AUC'.",
"FDtau"," (required only when diversity = 'FD' and FDtype = 'tau_values'), a numerical vector between 0 and 1 specifying tau values (threshold levels). If NULL (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy)."
)
output <- 
  matrix(Des, 
         ncol=2, byrow = TRUE)

library(htmlTable)
htmlTable(output,
          header =  c("Argument","Description"),align = "l"
          )
```

the `iNEXTbeta.link()` function returns the `"iNEXTbeta.link"` object including seven data frames for each regions: 

- gamma 
- alpha 
- beta
- C ( Sorensen-type non-overlap )
- U ( Jaccard-type  non-overlap )
- V ( Sorensen-type turnover )
- S ( Jaccard-type  turnover ) 


```{r eval=FALSE}
# Taxonomic diversity
Abundance_TD = iNEXTbeta.link(data = beetles, diversity = 'TD', q = c(0, 1, 2))
Abundance_TD
```


```{r echo=FALSE}
Abundance_TD = iNEXTbeta.link(data = beetles, diversity = 'TD', level = seq(0.5,1,0.1), q = c(0, 1, 2))
lapply(Abundance_TD$Region_1, function(x) {
    tmp = x[1:6,]
    tmp[,c('Estimate', 'SC','Size', 's.e.', 'LCL', 'UCL')] = round(tmp[,c('Estimate', 'SC','Size', 's.e.', 'LCL', 'UCL')], 3)
    tmp
})
```

The output contains seven data frames: `gamma`, `alpha`, `beta`, `C`, `U`, `V`, `S`. For each data frame, it includes the diversity estimate (`Estimate`), the diversity order (`Order.q`), `Method` (Rarefaction, Observed, or Extrapolation, depending on whether the size `m` is less than, equal to, or greater than the reference sample size), the sample coverage estimate (`SC`), the sample size (`Size`), the standard error from bootstrap replications (`s.e.`), the 95% lower and upper confidence limits of diversity (`LCL`, `UCL`), and the name of region (`Region`). These diversity estimates with confidence intervals are used for plotting the diversity curve.

## GRAPHIC DISPLAYS FUNCTION: ggiNEXTbeta.link()

The function `ggiNEXTbeta.link()`, which extends `ggplot2` to the `"iNEXT.link"` object with default arguments, is described as follows: 


```{r, echo=FALSE,warning=FALSE}
Des <- c("output"," the output of <code>iNEXTbeta.link</code>.",
"type"," selection of plot type : type = 'B' for plotting the gamma, alpha, and beta diversity; type = 'D' for plotting 4 turnover dissimilarities.",
"scale"," Are scales shared across all facets (the default, 'fixed'), or do they vary across rows ('free_x'), columns ('free_y'), or both rows and columns ('free')? Default is 'free'."
)
output <- 
  matrix(Des, 
         ncol=2, byrow = TRUE)

library(htmlTable)
htmlTable(output,
          header =  c("Argument","Description"),align = "l"
          )
```


The `ggiNEXTbeta.link()` function is a wrapper around the `ggplot2` package to create a R/E curve using a single line of code. The resulting object is of class `"ggplot"`, so it can be manipulated using the `ggplot2` tools. Users can visualize the output of beta diversity or four dissimilarities by setting the parameter <code>**type**</code>:

```{r, fig.align='center', fig.height=6, fig.width=6}
ggiNEXTbeta.link(Abundance_TD, type = 'B')
```

```{r, fig.align='center', fig.height=8, fig.width=6}
ggiNEXTbeta.link(Abundance_TD, type = 'D')
```



### How to cite
- Chiu, C-H., Chao, A., Vogel, S., Kriegel, P. and Thorn, S. (2023). Network-diversity quantification and related statistical estimation: drawing on sampling models and methodologies from biodiversity research. To appear in Philosophical Transactions of the Royal Society B.


### Referance

Chiu, C-H., Chao, A., Vogel, S., Kriegel, P. and Thorn, S. (2023). Network-diversity quantification and related statistical estimation: drawing on sampling models and methodologies from biodiversity research. To appear in Philosophical Transactions of the Royal Society B.


